### ISSUE TITLE
Fix a11y failure for best practice on landmark names

### ISSUE THREAD

**Issue by maureenlholland** [mozmeao/springfield#988]
## Description

Flagged on desktop release notes with pagination: https://www.firefox.com/en-US/firefox/147.0/releasenotes/

Landmarks should have a unique role or role/label/title (i.e. accessible name) combination

```
<nav class="c-release-pagination mzp-l-content" aria-label="Release notes pagination">
```
---

#### Success Criteria
- [ ] Unique landmark name and a11y tests pass
